index.js

```diff
@@ -68,11 +68,21 @@ function adapter(uri, opts){
 
     this.uid = uid;
     this.prefix = prefix;
+    this.channel = prefix + '#' + nsp.name + '#';
+    if (String.prototype.startsWith) {
+      this.channelMatches = function (messageChannel, subscribedChannel) {
+        return messageChannel.startsWith(subscribedChannel);
+      }
+    } else { // Fallback to slow indexOf impl for older Node.js
+      this.channelMatches = function (messageChannel, subscribedChannel) {
+        return messageChannel.indexOf(subscribedChannel) === 0;
+      }
```

Comment:
@rauchg I referred to https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/String/startsWith

I leant something from your suggestion, thank you!